test/integration/discovery.go

```diff
@@ -306,13 +309,161 @@ var _ = Describe("Discover Cluster", func() {
+			var zoneLabelValue = "us-west-2"
+			var regionLabelValue = "us-west-2"
+
+			zoneFlag := make(map[string]bool)
+			zoneFlag["HonorRegionZoneLabels"] = false
+			err = utilfeature.DefaultMutableFeatureGate.SetFromMap(zoneFlag)
+			if err != nil {
+				glog.Fatalf("Invalid Feature Gates: %v", err)
+			}
+
+			pvc, err := createVolumeClaim(kubeClient, namespace, "standard")
+			if pvc == nil && err != nil {
+				framework.Failf("Failed to create PVC %s with the error %s", pvc, err)
+			}
+
+			//Create a deployment with volume
+			dep, err := createDeployResource(kubeClient, depSingleContainerWithResources(namespace, pvc.Name, 1, true, false, false, ""))
+			framework.ExpectNoError(err, "Error creating test resources")
+
+			pod, err := getPodWithNamePrefix(kubeClient, dep.Name, namespace, "")
+			framework.ExpectNoError(err, "Error getting deployments pod")
+			// This should not happen. We should ideally get a pod.
+			if pod == nil {
+				framework.Failf("Failed to find a pod for deployment: %s", dep.Name)
+			}
+
+			nodeName = pod.Spec.NodeName
+			zoneNode, err = kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			if err != nil {
+				framework.Failf("Failed to retrieve node with error %s", err)
+			}
+			zoneNode.Labels[zoneLabelKey] = zoneLabelValue //Label node with zone labels
+			zoneNode.Labels[regionLabelKey] = regionLabelValue
+			zoneNode, err = kubeClient.CoreV1().Nodes().Update(context.TODO(), zoneNode, metav1.UpdateOptions{})
+			if err != nil {
+				framework.Failf("Failed to update node %s with zone label: %s %s", zoneNode, zoneLabelKey+":"+zoneLabelValue, err)
+			}
+
+			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
+			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
+
+			podFullName = namespace + "/" + pod.Name
+			commodityZoneValue = zoneLabelKey + "=" + zoneLabelValue
+			commodityRegionValue = regionLabelKey + "=" + regionLabelValue
+
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
+
+			//Verify zone/region label in the commodity list
+			if validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityZoneValue) {
+				//Teardown Remove both labels
+				deleteNode := deleteTopologyLabels(zoneNode, zoneLabelKey, kubeClient)
+				_ = deleteTopologyLabels(deleteNode, regionLabelKey, kubeClient)
+				framework.Failf("Zone label in node for pod with PV is not honored")
+			}
+			if validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityRegionValue) {
+				deleteNode := deleteTopologyLabels(zoneNode, zoneLabelKey, kubeClient)
+				_ = deleteTopologyLabels(deleteNode, regionLabelKey, kubeClient)
+				framework.Failf("Region label in node for pod with PV is not honored")
+			}
+			//Teardown Remove both labels
+			deleteNode := deleteTopologyLabels(zoneNode, zoneLabelKey, kubeClient)
```

Comment:
Better add `AfterEach` and put the `deleteTopologyLabels` there, in case the test failed and there is no chance to run this cleanup.